src/App.tsx

```diff
@@ -1,223 +1,188 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+  useMemo,
+} from 'react';
+import { addTodo, deleteTodo, getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+import { ErrorType } from './types/ErrorType';
+import { FilterBy } from './types/FilterBy';
+import { getVisibletodos } from './utils/getVisibleTodos';
+import { NewTodoField } from './components/NewTodoField';
+import { TodosCountInfo } from './components/TodosCountInfo/TodosCountInfo';
+import { ClearCompletedTodos } from './components/ClearCompletedTodos';
+import { TodosFilter } from './components/TodosFilter';
+import { TodosList } from './components/TodosList';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TempTodo } from './components/TempTodo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.ALL);
+  const [error, setError] = useState<ErrorType>(ErrorType.NONE);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [isDeleting, setIsDeleting] = useState(false);
+  const [deletingIds, setDeletingIds] = useState<number[]>([]);
+
+  const loadTodos = useCallback(async () => {
+    if (user) {
+      try {
+        const TodosFromApi = await getTodos(user?.id);
+
+        setTodos(TodosFromApi);
+      } catch {
+        throw new Error('Todos not found');
+      }
+    }
+  }, [user]);
+
+  useEffect(() => {
+    loadTodos();
+  }, [isAdding, isDeleting]);
+
+  useEffect(() => {
+    setTimeout(() => setError(ErrorType.NONE), 3000);
+  }, [error]);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    setError(ErrorType.NONE);
+  }, [filterBy]);
+
+  const handleError = (errorType: ErrorType) => {
+    setError(errorType);
+  };
+
+  const handleAddTodo = useCallback(async (
+    event: React.FormEvent<HTMLFormElement>,
+  ) => {
+    event.preventDefault();
+
+    if (!newTodoTitle.trim()) {
+      setError(ErrorType.EMPTYTITLE);
+    }
+
+    if (user && newTodoTitle.trim()) {
+      setIsAdding(true);
+
+      const newTodo = {
+        userId: user.id,
+        title: newTodoTitle.trim(),
+        completed: false,
+      };
+
+      try {
+        await addTodo(newTodo);
+      } catch {
+        setError(ErrorType.ADD);
+      }
     }
+
+    setNewTodoTitle('');
+    setIsAdding(false);
```

Comment:
When adding a new todo, after the animation of loading to the server, the list  is gapping, maybe you should reload todo list before, you setting false in setIsAdding

Reply:
Good point, thanks!))